x/auth/ante/ante.go

```diff
@@ -20,8 +19,16 @@ type HandlerOptions struct {
 	SignModeHandler        *txsigning.HandlerMap
 	SigGasConsumer         func(meter storetypes.GasMeter, sig signing.SignatureV2, params types.Params) error
 	TxFeeChecker           TxFeeChecker
+	// TODO HV2: is FeeCollector == FeegrantKeeper?
+	//  Can our business logic be implemented by using the FeegrantKeeper and avoid additional changes?
+	//  Also, heimdall uses FeeCollector for its interface method `SendCoinsFromAccountToModule`.
+	//  Since supply module is merged with bank module upstream, is BankKeeper's method enough?
+	FeeCollector FeeCollector
 }
 
+// TODO HV2: We need to double check the NewAnteHandler function and all the Decorators it uses (especially the AnteHandle methods of each Decorator).
+//  Also, is this implementation enough or we need further changes to reconcile with heimdall's `auth/ante.go` ?
```

Comment:
I took a closer look at the changes in `auth/ante.go` in [current Heimdall](https://github.com/Raneet10/cosmos-sdk/pull/2). These are the changes I saw (other changes are irrelevant)

* Extra logic to check if milestone msgs can be executed
* `EnsureSufficientMempoolFees` and `ConsumeGas` with "txSize" removed
* Only 1 signer
* Big changes to `processSig` and `GetSignBytes`

If you compare the behavior of all `AnteHandle` and try to map them 1 to 1, you will probably find many changes because things have changed a lot between the two versions. But changes in `AnteHandle` logic that was not touched in HeimdallV1 should not matter.

IMO you should focus on the _real_ changes in logic (i.e., the list above): locate where they are in the new SDK code, and "transplant" the changes there.

Reply:
Yeah that is what I basically tried to do (please have a look at the decorators to confirm).
I extensively changed the places where `processSig` and `GetSignBytes` where called imo, and enforced the 1 signer check.
Maybe I only left out the milestone one as the whole feature is not present here yet.
Anyway, I'll double check once again all the decorators. 
Please do the same if you have some time, thanks

Reply:
Ok, rechecked everything here.
If I am not wrong, all the places have been located and the logic "transplanted".
The only thing missing was the milestone check, which i just imported.
Will anyway leave the `TODO HV2` comments to double check everything once migration is completed.

Reply:
I will try to re-check on my side today

Reply:
Back to `GetSignBytes`, there is a change in Heimdall that converts "newData" ("data":"0x") into "oldData" ("data":"0x0"). I don't seem to find this change in the PR. Is it needed?

Reply:
I'll be double checking this based on the original PR https://github.com/maticnetwork/heimdall/pull/1016

Reply:
We checked internally. As we're going with HF approach, we don't need this.
Thanks

Reply:
Resolved